cluster/discovery/discovery.go

```diff
@@ -0,0 +1,33 @@
+package discovery
+
+// NodeEntry is used to discovery nodes in the cluster
+type NodeEntry struct {
+	Id            string
+	Ip            string
+	GossipVersion string
```

Comment:
Currently we look at this GossipVersion and then decide if we should add it in our gossip map. Basically the list of Ips that we pass to gossip are decided based on gossip version matches.